src/validators/password.js

```diff
@@ -0,0 +1,81 @@
+import i18n from '@dhis2/d2-i18n'
+import { isEmpty, isString } from './helpers/index.js'
+
+const LOWER_CASE_PATTERN = /^(?=.*[a-z]).+$/
+const UPPER_CASE_PATTERN = /^(?=.*[A-Z]).+$/
+const DIGIT_PATTERN = /^(?=.*[0-9]).+$/
+// Using this regex to match all non-alphanumeric characters to match server-side implementation
+// https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/SpecialCharacterValidationRule.java#L39
+const SPECIAL_CHARACTER_PATTERN = /[^a-zA-Z0-9]/
+
+const notString = i18n.t('Password should be a string')
+const tooShort = i18n.t('Password should be at least 8 characters long')
+const tooLong = i18n.t('Password should be no longer than 34 characters')
+const noLowerCase = i18n.t(
+    'Password should contain at least one lowercase letter'
+)
+const noUpperCase = i18n.t(
+    'Password should contain at least one UPPERCASE letter'
+)
+const noNumber = i18n.t('Password should contain at least one number')
+const noSpecialCharacter = i18n.t(
+    'Password should have at least one special character'
+)
+
+/**
+ * Tests if a given password is compliant with the password restrictions.
+ * This function checks all restrictions below, but returns when the first violation was found:
+ * - At least 8 characters
+ * - No more than 34 characters
+ * - Contains at least 1 lowercase character
+ * - Contains at least 1 UPPERCASE character
+ * - Contains at least 1 number
+ * - Contains at least 1 special character
+ */
+const password = value => {
```

Comment:
I don't like these static checks here if we call this one just `password`, because it's very dhis2 specific. `ui-core` is meant to work outside of the dhis2 context. `ui-forms` is meant to be nothing but a bridge between `ui-core` and RFF, so I'd either call this one `dhis2Password` or make it accept either a config object to be able to adjust these values or accept a pattern (which would then be a duplicate of the `createEqualTo` validator and therefor unnecessary)

Reply:
I've renamed to dhis2Password in https://github.com/dhis2/ui-forms/pull/164/commits/2d2685b0b49a18d0b4d71be1cd8ac025ae3e4e23